src/vs/workbench/api/common/extHostChatAgents2.ts

```diff
@@ -198,20 +198,22 @@ class ChatAgentResponseStream {
 	}
 }
 
-export class ExtHostChatAgents2 implements ExtHostChatAgentsShape2 {
+export class ExtHostChatAgents2 extends Disposable implements ExtHostChatAgentsShape2 {
 
 	private static _idPool = 0;
 
 	private readonly _agents = new Map<number, ExtHostChatAgent>();
 	private readonly _proxy: MainThreadChatAgentsShape2;
 
-	private readonly _sessionDisposables: DisposableMap<string, DisposableStore> = new DisposableMap();
+	private readonly _sessionDisposables: DisposableMap<string, DisposableStore> = this._register(new DisposableMap());
+	private readonly _completionDisposables = this._register(new DisposableStore());
```

Comment:
Should be disposing these at some point

Reply:
Yeah, should `ExtHostChatAgents2` be torn down when the main thread proxy is torn down?

Reply:
It will- but these completion disposables are only valid for the lifetime of the completion. Instead of accumulating them throughout the lifetime of the window, which is essentially a leak, we should dispose them when we know they aren't valid anymore. That might be when the query that you're getting completions for has been submitted, or the next time you call the same completion provider.

Reply:
https://github.com/microsoft/vscode/pull/211490